File: infra/src/main/java/com/beat/infra/external/auth/social/kakao/response/KakaoUserProfile.java
package com.beat.infra.external.auth.social.kakao.response;

import com.fasterxml.jackson.annotation.JsonProperty;

public record KakaoUserProfile(
	@JsonProperty("nickname")
	String nickname,
	@JsonProperty("thumbnail_image_url")
	String thumbnailImageUrl,
	@JsonProperty("profile_image_url")
	String profileImageUrl,
	@JsonProperty("is_default_image")
	Boolean defaultImage
) {
}

File: core/infra/src/main/kotlin/com/beat/infra/external/social/kakao/response/KakaoUserProfile.kt
package com.beat.infra.external.social.kakao.response

import com.fasterxml.jackson.annotation.JsonProperty

data class KakaoUserProfile(
    @param:JsonProperty("nickname")
    val nickname: String?,
    @param:JsonProperty("thumbnail_image_url")
    val thumbnailImageUrl: String?,
    @param:JsonProperty("profile_image_url")
    val profileImageUrl: String?,
    @param:JsonProperty("is_default_image")
    val defaultImage: Boolean?,
)
